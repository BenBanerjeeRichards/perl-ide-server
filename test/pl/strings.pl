#!/usr/bin/perl

use feature "say";

say qq H\Hello World!H;
say qw@Hello \@ World@;
say q{Hello!};

say qr"Hello";
say qr{Hi!};
say qr(Helllo);
tr TlTxT;
say qq {Hello {World} \{};

s/Hello/World/;
s WTHingWAreW;
s{Hello}{Wor{}sld}i;

$subject =~ s{A} {B}gi;
$event   =~ s{C} {D}gi;

q {Hello};
q ZHEllo {} WorldZ;

tr {Hello}{World};
tr {Hello} {World};
tr {Hello} /World/;
tr {Hello}/World/;
tr /Hello/World/;


tr{Hello}<World>;
tr{Hello}!World!;

s /
    Hello
    World
   /
    Indeed!

/;

$program =~ s {
    /\*	# Match the opening delimiter.
        .*?	# Match a minimal number of characters.
        \*/	# Match the closing delimiter.
} []gsx;

tr///;
tr{}[];


my $x = 50;
#
sub clean_and_trim_subject {
    $$ref =~ s/\n.*//s;
}

my $event_eat = [qw[head title style layer applet object xml param base]];
my $event_remove = [qw[bgsound embed object link body meta noscript plaintext noframes]];

s WHelloWorldWsdxx;
qr /Hello/xs;
m /pattern/sxx;
m {Pattern}sxx;
/Hello/s;
tr/hello/world/cdsr;
y/world/helllo/cdsr;
s/PATTERN/REPLACEMENT/msixpodualngcer;
m/PATTERN/msixpodualngc;
/PATTERN/msixpodualngc;
qr/STRING/msixpodualn;



m/PATTERN/msixpongc;
/PATTERN/msixplngc;


q{\\};
q{\{};
q{\}};
