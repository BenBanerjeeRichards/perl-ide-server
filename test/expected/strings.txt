Comment(#!/usr/bin/perl)
Use
Name(feature)
String("say")
Semicolon
Name(say)
QuoteIdent(qq)
StringStart(H)
String(\Hello World!)
StringEnd(H)
Semicolon
Name(say)
QuoteIdent(qw)
StringStart(@)
String(Hello \@ World)
StringEnd(@)
Semicolon
Name(say)
QuoteIdent(q)
StringStart({)
String(Hello!)
StringEnd(})
Semicolon
Name(say)
QuoteIdent(qr)
StringStart(")
String(Hello)
StringEnd(")
Semicolon
Name(say)
QuoteIdent(qr)
StringStart({)
String(Hi!)
StringEnd(})
Semicolon
Name(say)
QuoteIdent(qr)
StringStart(()
String(Helllo)
StringEnd())
Semicolon
QuoteIdent(tr)
StringStart(T)
String(l)
StringEnd(T)
String(x)
StringEnd(T)
Semicolon
Name(say)
QuoteIdent(qq)
StringStart({)
String(Hello {World} \{)
StringEnd(})
Semicolon
QuoteIdent(s)
StringStart(/)
String(Hello)
StringEnd(/)
String(World)
StringEnd(/)
Semicolon
QuoteIdent(s)
StringStart(W)
String(THing)
StringEnd(W)
String(Are)
StringEnd(W)
Semicolon
QuoteIdent(s)
StringStart({)
String(Hello)
StringEnd(})
StringStart({)
String(Wor{}sld)
StringEnd(})
StringModifiers(i)
Semicolon
ScalarVariable($subject)
Operator(=~)
QuoteIdent(s)
StringStart({)
String(A)
StringEnd(})
StringStart({)
String(B)
StringEnd(})
StringModifiers(gi)
Semicolon
ScalarVariable($event)
Operator(=~)
QuoteIdent(s)
StringStart({)
String(C)
StringEnd(})
StringStart({)
String(D)
StringEnd(})
StringModifiers(gi)
Semicolon
QuoteIdent(q)
StringStart({)
String(Hello)
StringEnd(})
Semicolon
QuoteIdent(q)
StringStart(Z)
String(HEllo {} World)
StringEnd(Z)
Semicolon
QuoteIdent(tr)
StringStart({)
String(Hello)
StringEnd(})
StringStart({)
String(World)
StringEnd(})
Semicolon
QuoteIdent(tr)
StringStart({)
String(Hello)
StringEnd(})
StringStart({)
String(World)
StringEnd(})
Semicolon
QuoteIdent(tr)
StringStart({)
String(Hello)
StringEnd(})
StringStart(/)
String(World)
StringEnd(/)
Semicolon
QuoteIdent(tr)
StringStart({)
String(Hello)
StringEnd(})
StringStart(/)
String(World)
StringEnd(/)
Semicolon
QuoteIdent(tr)
StringStart(/)
String(Hello)
StringEnd(/)
String(World)
StringEnd(/)
Semicolon
QuoteIdent(tr)
StringStart({)
String(Hello)
StringEnd(})
Comment(#World)
StringStart({)
String(World)
StringEnd(})
Semicolon
QuoteIdent(tr)
StringStart({)
String(Hello)
StringEnd(})
Comment(#World)
Comment(# World!)
StringStart({)
String(World)
StringEnd(})
Semicolon
QuoteIdent(tr)
StringStart({)
String(Hello)
StringEnd(})
StringStart(<)
String(World)
StringEnd(>)
Semicolon
QuoteIdent(tr)
StringStart({)
String(Hello)
StringEnd(})
StringStart(!)
String(World)
StringEnd(!)
Semicolon
QuoteIdent(s)
StringStart(/)
String(\n    Hello\n    World\n   )
StringEnd(/)
String(\n    Indeed!\n\n)
StringEnd(/)
Semicolon
ScalarVariable($program)
Operator(=~)
QuoteIdent(s)
StringStart({)
String(\n    /\*	# Match the opening delimiter.\n        .*?	# Match a minimal number of characters.\n        \*/	# Match the closing delimiter.\n)
StringEnd(})
StringStart([)
StringEnd(])
StringModifiers(gsx)
Semicolon
QuoteIdent(tr)
StringStart(/)
StringEnd(/)
StringEnd(/)
Semicolon
QuoteIdent(tr)
StringStart({)
StringEnd(})
StringStart([)
StringEnd(])
Semicolon
My
ScalarVariable($x)
Assignment
NumericLiteral(50)
Semicolon
Comment(#)
Sub
SubName(clean_and_trim_subject)
LBracket({)
Deref($)
ScalarVariable($ref)
Operator(=~)
QuoteIdent(s)
StringStart(/)
String(\n.*)
StringEnd(/)
StringEnd(/)
StringModifiers(s)
Semicolon
RBracket
My
ScalarVariable($event_eat)
Assignment
LSquareBracket
QuoteIdent(qw)
StringStart([)
String(head title style layer applet object xml param base)
StringEnd(])
RSquareBracket
Semicolon
My
ScalarVariable($event_remove)
Assignment
LSquareBracket
QuoteIdent(qw)
StringStart([)
String(bgsound embed object link body meta noscript plaintext noframes)
StringEnd(])
RSquareBracket
Semicolon
QuoteIdent(s)
StringStart(W)
String(Hello)
StringEnd(W)
String(orld)
StringEnd(W)
StringModifiers(sdxx)
Semicolon
QuoteIdent(qr)
StringStart(/)
String(Hello)
StringEnd(/)
StringModifiers(xs)
Semicolon
QuoteIdent(m)
StringStart(/)
String(pattern)
StringEnd(/)
StringModifiers(sxx)
Semicolon
QuoteIdent(m)
StringStart({)
String(Pattern)
StringEnd(})
StringModifiers(sxx)
Semicolon
String(/Hello/)
StringModifiers(s)
Semicolon
QuoteIdent(tr)
StringStart(/)
String(hello)
StringEnd(/)
String(world)
StringEnd(/)
StringModifiers(cdsr)
Semicolon
QuoteIdent(y)
StringStart(/)
String(world)
StringEnd(/)
String(helllo)
StringEnd(/)
StringModifiers(cdsr)
Semicolon
QuoteIdent(s)
StringStart(/)
String(PATTERN)
StringEnd(/)
String(REPLACEMENT)
StringEnd(/)
StringModifiers(msixpodualngcer)
Semicolon
QuoteIdent(m)
StringStart(/)
String(PATTERN)
StringEnd(/)
StringModifiers(msixpodualngc)
Semicolon
String(/PATTERN/)
StringModifiers(msixpodualngc)
Semicolon
QuoteIdent(qr)
StringStart(/)
String(STRING)
StringEnd(/)
StringModifiers(msixpodualn)
Semicolon
QuoteIdent(m)
StringStart(/)
String(PATTERN)
StringEnd(/)
StringModifiers(msixpongc)
Semicolon
String(/PATTERN/)
StringModifiers(msixplngc)
Semicolon
QuoteIdent(q)
StringStart({)
String(\)
StringEnd(})
Semicolon
QuoteIdent(q)
StringStart({)
String(\{)
StringEnd(})
Semicolon
QuoteIdent(q)
StringStart({)
String(\})
StringEnd(})
Semicolon
QuoteIdent(s)
Comment(# Test)
Comment(# World)
StringStart({)
String(^([^\r\n]*)You \s+ should \s+ have \s+ received\n    .*? (write \s+ to | contact .*? Foundation)\n)
StringEnd(})
Comment(# Comment!)
Comment(# Comment!)
StringStart({)
String(x)
StringEnd(})
StringModifiers(smix)
Semicolon
Name(say)
QuoteIdent(s)
Comment(#WOrld)
Comment(#Hello)
StringStart({)
String(Hello)
StringEnd(})
StringStart({)
String(World)
StringEnd(})
Semicolon
Name(say)
QuoteIdent(qq)
StringStart(W)
String(Hello)
StringEnd(W)
Semicolon
EndOfInput
